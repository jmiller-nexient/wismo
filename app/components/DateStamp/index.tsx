import * as React from 'react';

interface IDateStampProps {
  date: string;
  short?: boolean;
}

const formatDateLong = (date: string): string => {
  const [year, month, day] = date.split(/[T/-]/);
  const d = new Date(`${year}, ${month}, ${day}`);
  const days = ['Sunday', 'Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday'];
  const months = ['Jan.', 'Feb.', 'March', 'April', 'May', 'June', 'July', 'Aug.', 'Sep.', 'Oct.', 'Nov.', 'Dec.'];

  return `${days[d.getDay()]}, ${months[+month - 1]} ${day}`;
};

const formatDateShort = (date: string): string => {
  const [year, month, day] = date.split(/[T/-]/);

  return `${month}/${day}/${year}`;
};

const isValid = (date: string): boolean => {
  const d = Date.parse(date);

  return isNaN(d) === false;
};

const DateStamp: React.FC<IDateStampProps> = ({ date, short = false }: IDateStampProps): React.ReactElement => {
  if (isValid(date)) {
    if (short) {
      return <>{formatDateShort(date)}</>;
    }

    return <>{formatDateLong(date)}</>;
  }

  return <></>;
};

export default DateStamp;
