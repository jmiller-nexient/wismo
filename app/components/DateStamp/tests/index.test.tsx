import React from 'react';
import renderer from 'react-test-renderer';

import DateStamp from '..';

describe('DateStamp Component', (): void => {
  it('should render long datestamp when given ISO formatted date prop', (): void => {
    const tree = renderer.create(
      <DateStamp date={'2020-03-19T13:15:30Z'} />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render long datestamp when short prop is false', (): void => {
    const tree = renderer.create(
      <DateStamp date={'2020/04/10'} short={false}/>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render long datestamp when short prop is not given', (): void => {
    const tree = renderer.create(
      <DateStamp date={'2020/04/10'} />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render nothing when given invalid date prop', (): void => {
    const tree = renderer.create(
      <DateStamp date={'Tuesday'} />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render short datestamp when short prop is true', (): void => {
    const tree = renderer.create(
      <DateStamp date={'2020/04/10'} short={true}/>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render short datestamp when short prop is true and given ISO formatted date prop', (): void => {
    const tree = renderer.create(
      <DateStamp date={'2020-03-19T13:15:30Z'} short={true}/>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
