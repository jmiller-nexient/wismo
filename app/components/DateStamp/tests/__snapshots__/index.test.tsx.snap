// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateStamp Component should render long datestamp when given ISO formatted date prop 1`] = `"Thursday, March 19"`;

exports[`DateStamp Component should render long datestamp when short prop is false 1`] = `"Friday, April 10"`;

exports[`DateStamp Component should render long datestamp when short prop is not given 1`] = `"Friday, April 10"`;

exports[`DateStamp Component should render nothing when given invalid date prop 1`] = `null`;

exports[`DateStamp Component should render short datestamp when short prop is true 1`] = `"04/10/2020"`;

exports[`DateStamp Component should render short datestamp when short prop is true and given ISO formatted date prop 1`] = `"03/19/2020"`;
